### ISSUE TITLE
ModuleRepositoryTest has tests without any assertion

### ISSUE THREAD

**Issue by mvorisek** [PrestaShop/PrestaShop#23978]
#### Describe the bug

```
Run composer run-script phpunit-legacy --timeout=0
> @php ./tests-legacy/create-test-db.php
> @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests-legacy/phpunit.xml
...

There were 6 risky tests:

1) LegacyTests\Unit\Core\Addon\Module\ModuleRepositoryTest::testGetOnlyInstalledModules
This test did not perform any assertions

/home/runner/work/PrestaShop/PrestaShop/tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php:161

2) LegacyTests\Unit\Core\Addon\Module\ModuleRepositoryTest::testGetOnlyEnabledModules
This test did not perform any assertions

/home/runner/work/PrestaShop/PrestaShop/tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php:207

3) LegacyTests\Unit\Core\Addon\Module\ModuleRepositoryTest::testGetNotEnabledModules
This test did not perform any assertions

/home/runner/work/PrestaShop/PrestaShop/tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php:232

4) LegacyTests\Unit\Core\Addon\Module\ModuleRepositoryTest::testGetInstalledButDisabledModules
This test did not perform any assertions

/home/runner/work/PrestaShop/PrestaShop/tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php:254

5) LegacyTests\Unit\Core\Addon\Module\ModuleRepositoryTest::testGetAddonsFromMarketplaceOnly
This test did not perform any assertions

/home/runner/work/PrestaShop/PrestaShop/tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php:277

6) LegacyTests\Unit\Core\Addon\Module\ModuleRepositoryTest::testGetOnlyServices
This test did not perform any assertions

/home/runner/work/PrestaShop/PrestaShop/tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php:323
```

#### Expected behavior

The tests in `ModuleRepositoryTest` should perform an assertion. If no assertion is perfomed, I think nothing is tested, as assertions are presented in the code.

#### Steps to Reproduce

1. Open any `PHPUnit Tests (With DB) (7.4)` test and check the test results

#### Additional information

* PrestaShop version: develop

**Comment by sarahdib** [comment]
Hello @mvorisek,

Thank you for this report.

@PrestaShop/prestashop-core-developers this is a technical issue could you check please :) 

Thank you

**Comment by matks** [comment]
Hello @mvorisek you are right these tests have to be improved.

The content of `/tests-legacy` folder is old and not very good tests.

You can find the information in https://devdocs.prestashop.com/1.7/testing/introduction/#what-are-legacy-tests-in-tests-legacy-folder

The tests are scoped to be deleted one day, when they are replaced by better tests. But before this day arrives, we keep them running as they can still indicate some issues when we introduce changes.

> The tests in ModuleRepositoryTest should perform an assertion. If no assertion is perfomed, I think nothing is tested

One test without assertion can have value, which is "it ran successfully without crashing". Sometimes I add a `$this->assertTrue(true)` at the end of such tests 😆  to have at least one assertion.

Basically
- if the test is able to go until the last statement without crashing, it worked
- if the test crashes before, it failed

I agree this is no good practice but the tests there have been written a long time ago. They will be replaced by better tests in the future.

I close the issue as it's something already known and scoped 😉  .

**Comment by mvorisek** [comment]
There is `@doesNotPerformAssertions` phpdoc if this is desired, should be fixed to show the developer explicitly :)
